src/collections/src/NICollectionViewCellFactory.m

```diff
@@ -72,6 +79,14 @@ + (UICollectionViewCell *)cellWithNib:(UINib *)collectionViewCellNib
   UICollectionViewCell* cell = nil;
 
   NSString* identifier = NSStringFromClass([object class]);
+
+  if ([object respondsToSelector:@selector(reuseIdentifierSuffix)]) {
+    NSString* suffix = [object reuseIdentifierSuffix];
+    if (suffix.length) {
+      identifier = [identifier stringByAppendingFormat:@".%@", suffix];
+    }
+  }
```

Comment:
Part of me wonders if we should consider the nib case as a separate feature request and wait for someone to ask for support for that case before implementing support. With that said, the overhead doesn't seem particularly high to just go ahead and land support for it now.

Reply:
I don't see any real downside to just including this here as well. It will also guarantee consistency regardless of which initializer is used.